### ISSUE TITLE
Assembly: Reverse direction of the joint also inverts joints of sub-assembly

### ISSUE THREAD

**Issue by reinierstorm** [FreeCAD/FreeCAD#18187]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I am inserting a sub-assembly from a different file of a simple hinge:

![image](https://github.com/user-attachments/assets/84a98caa-76df-450f-89ec-9fef08554a6a)

When I join the hinge to another object it seems to invert the joint be default (The local coordinate axes all point in opposite directions):

![image](https://github.com/user-attachments/assets/c840b15a-da63-4a96-b882-3a583397b64a)

When inverting the joint all internal joints of the sub-assembly get inverted:

![image](https://github.com/user-attachments/assets/fc218340-e442-49a0-b942-bcfe415f13da)

These are the two files used:
[hinge_assemlby.zip](https://github.com/user-attachments/files/17957824/hinge_assemlby.zip)

### Full version info

```shell
OS: Arch Linux (i3/i3/xcb)
Architecture: x86_64
Version: 1.0.0.39109 (Git)
Build type: Release
Branch: makepkg
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.12.7, Qt 6.8.0, Coin 4.0.3, Vtk 9.3.1, OCC 7.8.1
Locale: C/Default (C)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/
Installed mods: 
  * joints.git
  * graph_test
  * Dracula 0.0.5
  * Behave-Dark-Colors 0.1.1
  * DynamicData 2.68.0
  * freecad.gears 1.3.0
  * Render 2024.9.25
  * sheetmetal 0.5.3
  * Nodes 0.1.36
  * fasteners 0.5.29
  * ExtremeProDark 2.9.0
  * Joints
```

### Subproject(s) affected?

Assembly

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by FEA-eng** [comment]
Still relevant

```
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.14555 (Git shallow)
Build date: 2025/09/09 06:43:51
Build type: Release
Branch: main
Hash: c1dd116e1e3304511d9e5e634e711e549b2b84e7
Python 3.11.13, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Navigation Style/Orbit Style/Rotation Mode: Blender/Rounded Arcball/Window center
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Logical DPI/Physical DPI/Pixel Ratio: 96/81.5973/1
```

**Comment by PaddleStroke** [comment]
It would have been best to have precise reproductions steps. Or video showing the issue. I'm struggling a bit finding out the exact issue and sequence to get it

**Comment by PaddleStroke** [comment]
https://github.com/user-attachments/assets/939fd78d-a15d-42f9-bfc6-4809d9019c1a

I played a bit with the file and cannot find any issue.

**Comment by FEA-eng** [comment]
> It would have been best to have precise reproductions steps. Or video showing the issue. I'm struggling a bit finding out the exact issue and sequence to get it

Right, I thought I've seen some affected subassembly joints while inverting the global assembly ones but I'm not sure how to properly reproduce it either.

**Comment by PaddleStroke** [comment]
One feature request I can see is that when you have a chain of joints like : 
PartA (grounded) -(joint1)> PartB -(joint2)>partC -(joint3)> partD

ie when you don't have a loop, ie PartB is only connected to ground through joint1.
Then reversing joint1 could move all the subparts (C and D) so that the whole block B C D reverse entirely.
